### PR TITLE
allow public video privacy to be deleted in the web client

### LINKED ISSUE #4159: Deleting privacy options breaks video uploading
**Describe the current behavior**
I've created a very basic plugin that all it does is this:
```
  registerHook({
    target: 'action:application.listening',
    handler: () => {
      videoPrivacyManager.deletePrivacy(1);
      videoPrivacyManager.deletePrivacy(2);
      videoPrivacyManager.deletePrivacy(3);
    }
  })
```

Context: I'd like a 'family' peertube which only Internal (4) videos can be uploaded.

In doing the above, the video upload page breaks in multiple ways, most commonly, I cannot publish videos as the Privacy dropdown isn't filled out/cannot be selected properly/results in blank dropdown.

**Steps to reproduce:**

1. Create a basic plugin, with just the above handler
2. Naviate to the 'Publish Video' page and observe the Privacy

Screen Shot 2021-06-04 at 11 08 35 pm

**Describe the expected behavior**

I'd expect only 'Internal' is available to be selected, and the Dropdown box just contains that.

**Additional information**

* PeerTube instance:
  * URL: Private/internal
  * version: 3.2.1
  * NodeJS version: docker buster-production
  * Ffmpeg version: docker buster-production

* Browser name, version and platforms on which you could reproduce the bug: Firefox 89 Mac
* Link to browser console log if relevant:
* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`):

### PR DESCRIPTION
**Pull request by rigelk**
## Description

A small fix that factorises the static default video privacy to the function returning filtered video privacies, and computes the default video privacy based on those returned by the API.

## Related issues

fixes #4159

## Has this been tested?

- [x] 👍 yes, light tests using the feature

### CODE REVIEW

File: client/src/app/shared/shared-main/video/video.service.ts [hunk 1]
```diff
@@ -398,9 +398,14 @@ export class VideoService implements VideosProvider {
       }
     ]
 
-    return base
+    const videoPrivacies = base
       .filter(o => !!privacies.find(p => p.id === o.id)) // filter down to privacies that where in the input
       .map(o => ({ ...privacies[o.id - 1], ...o })) // merge the input privacies that contain a label, and extend them with a description
+
+    return {
+      firstStepPrivacyId: videoPrivacies.find(p => p.id === VideoPrivacy.PUBLIC).id || videoPrivacies[0].id,
```
Comment (Chocobozzz) [inline client/src/app/shared/shared-main/video/video.service.ts:406]: Put `VideoPrivacy.PUBLIC` as a default arg so we explicitly choose the default privacy in components

Comment (Chocobozzz) [reply client/src/app/shared/shared-main/video/video.service.ts:406]: `firstStepPrivacyId` -> `defaultPrivacyId`

**Review by Chocobozzz — CHANGES_REQUESTED** [review]

File: client/src/app/shared/shared-main/video/video.service.ts [hunk 2]
```diff
@@ -398,9 +398,14 @@ export class VideoService implements VideosProvider {
       }
     ]
 
-    return base
+    const videoPrivacies = base
       .filter(o => !!privacies.find(p => p.id === o.id)) // filter down to privacies that where in the input
       .map(o => ({ ...privacies[o.id - 1], ...o })) // merge the input privacies that contain a label, and extend them with a description
+
+    return {
+      defaultPrivacyId: videoPrivacies.find(p => p.id === defaultPrivacyId).id || videoPrivacies[0].id,
```
Comment (Chocobozzz) [inline client/src/app/shared/shared-main/video/video.service.ts:406]: Have you tested this code? Because `videoPrivacies.find` can return null

Comment (rigelk) [reply client/src/app/shared/shared-main/video/video.service.ts]: I corrected the attribute access.

**Pull request merged**